src/anonymous-credentials/types-and-consts.ts

```diff
@@ -57,3 +83,40 @@ export const ACCUMULATOR_PROVING_KEY_LABEL_BYTES = te.encode(ACCUMULATOR_PROVING
 // Label used for generating SAVER encryption generators
 export const SAVER_ENCRYPTION_GENS_LABEL = 'DockSAVEREncryptionGens2022';
 export const SAVER_ENCRYPTION_GENS_BYTES = te.encode(SAVER_ENCRYPTION_GENS_LABEL);
+
+function once<T>(f: () => T) {
+  const NO_VALUE = {};
+  let calculated: typeof NO_VALUE | T = NO_VALUE;
+
+  return function () {
+    if (calculated === NO_VALUE) {
+      return (calculated = f());
+    } else {
+      return calculated;
+    }
+  };
+}
+
+export const DEFAULT_SIGNATURE_LABEL_BYTES = Object.setPrototypeOf(
```

Comment:
Why not just 
```
DEFAULT_SIGNATURE_LABEL_BYTES = { 
   [BBSSignatureParams.name]: BBS_SIGNATURE_PARAMS_LABEL_BYTES , 
  ..
}
```

Reply:
Because of the object prototype properties.

Reply:
Can you please elaborate?

Reply:
```
Object.getOwnPropertyNames({}.__proto__) =>
(12) ['constructor', '__defineGetter__', '__defineSetter__', 'hasOwnProperty', '__lookupGetter__', '__lookupSetter__', 'isPrototypeOf', 'propertyIsEnumerable', 'toString', 'valueOf', '__proto__', 'toLocaleString']
```
When using an object as a map it's always better to nullify the prototype in order to avoid cases like `({})['constructor']` etc. While the current call might not be the case, any function which takes user input and then attempts to take a value from this map might be vulnerable.

Reply:
This (unlikely) problem exists at so many places in this codebase :smile: . Anyway, thanks for explaining.